Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp

```diff
@@ -225,44 +225,46 @@ void WebFoundTextRangeController::clearAllDecoratedFoundText()
 {
     clearCachedRanges();
     m_decoratedRanges.clear();
-    m_webPage->corePage()->unmarkAllTextMatches();
+    m_webPage->protectedCorePage()->unmarkAllTextMatches();
 
     m_highlightedRange = { };
     m_textIndicator = nullptr;
 
-    if (m_findPageOverlay)
-        m_findPageOverlay->setNeedsDisplay();
+    if (RefPtr findPageOverlay = m_findPageOverlay)
+        findPageOverlay->setNeedsDisplay();
 }
 
 void WebFoundTextRangeController::didBeginTextSearchOperation()
 {
-    if (m_findPageOverlay)
-        m_findPageOverlay->stopFadeOutAnimation();
+    if (RefPtr existingFindPageOverlay = m_findPageOverlay)
+        existingFindPageOverlay->stopFadeOutAnimation();
     else {
-        m_findPageOverlay = WebCore::PageOverlay::create(*this, WebCore::PageOverlay::OverlayType::Document);
-        m_webPage->corePage()->pageOverlayController().installPageOverlay(*m_findPageOverlay, WebCore::PageOverlay::FadeMode::Fade);
+        RefPtr findPageOverlay = WebCore::PageOverlay::create(*this, WebCore::PageOverlay::OverlayType::Document);
+        m_webPage->protectedCorePage()->pageOverlayController().installPageOverlay(*m_findPageOverlay, WebCore::PageOverlay::FadeMode::Fade);
+        m_findPageOverlay = WTFMove(findPageOverlay);
     }
 
-    m_findPageOverlay->setNeedsDisplay();
+    RefPtr { m_findPageOverlay }->setNeedsDisplay();
```

Comment:
`RefPtr { m_findPageOverlay }` is used in several places, we should introduce a `protectedFindPageOverlay()` function instead.